tests/integration_tests/ui_tests/ArelleGUITest/ArelleGUITest/Tests.cs

```diff
@@ -0,0 +1,373 @@
+        public void TestLoadDocument()
+        {
+            Logger.Default.Info("start TestLoadDocument");
+            Keyboard.TypeSimultaneously(VirtualKeyShort.CONTROL, VirtualKeyShort.KEY_O);
+            Wait.UntilInputIsProcessed();
+            Logger.Default.Info("find the Open File dialog");
+            // more reliable than Window.ModalWindows because that searches all descendants whereas this only searches children
+            var openFileWindow = Retry.WhileNull(() => Window.FindFirstChild(cf =>
+                cf.ByControlType(ControlType.Window).And(new PropertyCondition(Automation.PropertyLibrary.Window.IsModal, true))),
+                timeout: TimeSpan.FromSeconds(5)).Result;
+            Assert.That(openFileWindow, Is.Not.Null);
+
+            var exampleDocumentPath = Path.Join(ExecutionType.ArelleResourcesPath, "workiva.zip");
+            var exampleDocumentEntryPoint = "wk-20220331.htm";
+            int nFilesInDocumentZip, indexOfEntryPoint;
+            using (var archive = ZipFile.OpenRead(exampleDocumentPath))
+            {
+                nFilesInDocumentZip = archive.Entries.Count;
+                indexOfEntryPoint = archive.Entries.Select((e, i) => new { e, i }).Where(x => x.e.Name == exampleDocumentEntryPoint).Single().i;
+            }
+            var fileNameComboBox = Retry.WhileNull(
+                () => openFileWindow.FindFirstDescendant(cf => cf.ByControlType(ControlType.ComboBox).And(cf.ByName("File name:")))).Result.AsComboBox();
+            Logger.Default.Info("enter path to document");
+            fileNameComboBox.EditableText = exampleDocumentPath;
+            Wait.UntilInputIsProcessed();
+            Logger.Default.Info("open document");
+            Keyboard.Type(VirtualKeyShort.ENTER);
+            Wait.UntilInputIsProcessed();
+
+            Logger.Default.Info("select file inside document archive");
+            var selectFileFromArchiveWindow = Retry.WhileNull(
+                () => Window.FindFirstChild(Automation.ConditionFactory.ByControlType(ControlType.Window)), timeout: TimeSpan.FromSeconds(3)).Result;
+            var selectFileFromArchivePane = selectFileFromArchiveWindow.FindFirstByXPath("/Pane/Pane/Pane/Pane[1]");
+            Mouse.LeftClick(new Point(
+                (int)(selectFileFromArchivePane.BoundingRectangle.X + selectFileFromArchivePane.BoundingRectangle.Width / 2.0f),
+                (int)(selectFileFromArchivePane.BoundingRectangle.Y
+                // +1 for the header, +0.5 to end up in the middle
+                + selectFileFromArchivePane.BoundingRectangle.Height * (indexOfEntryPoint + 1.5) / (nFilesInDocumentZip + 1))
+            ));
+            Keyboard.Type(VirtualKeyShort.TAB, VirtualKeyShort.ENTER);
+
+            Logger.Default.Info("wait for document to load and UI to populate");
+            Retry.WhileNull(() => Window.FindFirstByXPath("/Pane/Pane/Pane/Pane/Pane/Pane/Pane"), timeout: TimeSpan.FromMinutes(1), ignoreException: true);

```

Comment:
What's this about?

Reply:
Maybe `waitForNestedPane(depth: 7)`?

Reply:
A comment explaining what in the UI this is looking for would be appreciated (and help support debugging if it breaks).